### DISCUSSION TITLE
GridData custom command button

### DISCUSSION THREAD

**Discussion by GilShalit** [Megabit/Blazorise#2530] [Q&A] [answered]
Hi, how can I add a non standard CommandButton?  I saw the sample [here](https://stackoverflow.com/a/66787696/149769) but I would prefer to have the button in the command button column.

I want the button to be used to move the row up or down in the grid, and I have the logic for that.

#PushingDataGridToTheLimit

**Comment by stsrki** [comment]
We still don't have a template for custom commands. But, you can use one of the existing ones to add a custom button.

```cshtml
<DataGridCommandColumn TItem="Employee" Width="170px">
    <EditCommandTemplate>
        <Button Color="Color.Primary" Clicked="@context.Clicked">@context.LocalizationString</Button>
        <Button Color="Color.Secondary">Your custom command</Button>
    </EditCommandTemplate>
</DataGridCommandColumn>
```

**Reply by GilShalit** [reply]
Thanks!  But how do I connect the new button to my logic?  using 
```
		<Button Color="Color.Secondary" Clicked="@(()=>OnUpButtonClicked(context))">Up</Button>
```

and 
```
	async Task OnUpButtonClicked(myClass model)
	{
	}
```

results in a curly line under context and the error message: ```Argument 1: cannot convert from 'Microsoft.AspNetCore.Components.Authorization.AuthenticationState' to 'myClass'```

**Comment by stsrki** [comment, marked as answer]
Change it to `async Task OnUpButtonClicked(EditCommandContext<myClass> context)`
